go/signedexchange/mice/mice.go

```diff
@@ -0,0 +1,57 @@
+package mice
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/binary"
+	"io"
+)
+
+// Encode encodes the given content buf to MICE (Merkle Integrity Content Encoding)
+// format.
+//
+// Encode returns MI header field parameter string and error if exists.
+//
+// Spec: https://tools.ietf.org/html/draft-thomson-http-mice-02
+func Encode(w io.Writer, buf []byte, recordSize int) (string, error) {
+	numRecords := (len(buf) + recordSize - 1) / recordSize
+	if len(buf) == 0 {
+		numRecords = 1
+	}
+
+	proofs := make([][]byte, numRecords)
+	for i := 0; i < numRecords; i++ {
+		rec := numRecords - i - 1
```

Comment:
Please comment what you're doing in this loop.

Reply:
Done.